### PR TITLE
108 contact examples and fixes

### LINKED ISSUE #108: Missing numpy import in collision examples
**Describe the bug**
Attempting to run the collision examples hosted [here](https://github.com/GazzolaLab/PyElastica/tree/master/examples/RodContactCase) returns an error that `np` is not defined. Seems like `numpy` is not imported.

**To Reproduce**
```bash
python3 rod_rod_contact_parallel_validation.py
```

**Environment**
Environment independent.

**Expected behavior**
No bug when running the examples.

### PR DESCRIPTION
**Pull request by armantekinalp**
Fixes #108 and also we added example cases for rod-cylinder contact.

### CODE REVIEW

File: examples/RigidBodyCases/RodRigidBodyContact/rod_cylinder_contact_validation.py [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+
+inclination = np.deg2rad(30)
+direction = np.array([0.0, np.cos(inclination), np.sin(inclination)])
+normal = np.array([0.0, -np.sin(inclination), np.cos(inclination)])
+
+# can be y or z too, meant for testing purposes of rod-body contact in different planes
+action_plane_key = "x"
+
+# can be set to True, checks collision at tips of rod
+tip_collision = True
```
Comment (skim0119) [inline examples/RigidBodyCases/RodRigidBodyContact/rod_cylinder_contact_validation.py:29]: Capitalize flag?

File: examples/RigidBodyCases/RodRigidBodyContact/rod_cylinder_contact_validation.py [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+timestepper = PositionVerlet()
+final_time = 2.0
+dl = base_length / n_elem
+dt = 1e-4
+total_steps = int(final_time / dt)
+print("Total steps", total_steps)
+
+integrate(timestepper, single_rod_sim, final_time, total_steps)
+
+if True:
```
Comment (skim0119) [inline examples/RigidBodyCases/RodRigidBodyContact/rod_cylinder_contact_validation.py:152]: Flag name?

File: examples/RigidBodyCases/RodRigidBodyContact/rod_cylinder_contact_validation.py [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+        x_grid = cradius * np.cos(theta_grid) + center_x
+        y_grid = cradius * np.sin(theta_grid) + center_y
+        z_grid += cstart[2]
+        return np.roll([x_grid, y_grid, z_grid], _roll_key)
+
+    XC, YC, ZC = make_data_for_cylinder_along_z(
+        cylinder_start, cylinder_radius, cylinder_height
+    )
+
+    def plot_video(
```
Comment (skim0119) [inline examples/RigidBodyCases/RodRigidBodyContact/rod_cylinder_contact_validation.py:168]: These should be in `post_processing.py`, right?

File: examples/RigidBodyCases/RodRigidBodyContact/rod_cylinder_contact_validation.py [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+                        rod_line.set_xdata(positions[i, 0])
+                        rod_line.set_ydata(positions[i, 1])
+                        rod_line.set_3d_properties(positions[i, 2])
+                        XC, YC, ZC = make_data_for_cylinder_along_z(
+                            cylinder_origin[i, ...], cylinder_radius, cylinder_height
+                        )
+                        surf.remove()
+                        surf = ax.plot_surface(XC, YC, ZC, color="g", alpha=0.5)
+                        writer.grab_frame()
+        if True:
```
Comment (skim0119) [inline examples/RigidBodyCases/RodRigidBodyContact/rod_cylinder_contact_validation.py:238]: Flag name

File: examples/RigidBodyCases/RodRigidBodyContact/rod_cylinder_contact_with_y_normal.py [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+timestepper = PositionVerlet()
+final_time = 2.0
+dl = base_length / n_elem
+dt = 1e-4
+total_steps = int(final_time / dt)
+print("Total steps", total_steps)
+
+integrate(timestepper, single_rod_sim, final_time, total_steps)
+
+if True:
```
Comment (skim0119) [inline examples/RigidBodyCases/RodRigidBodyContact/rod_cylinder_contact_with_y_normal.py:130]: Flag name

**Review by skim0119 — CHANGES_REQUESTED** [review]

**Comment by armantekinalp** [comment]
Flags capitalized and post-processing scripts are moved to `post_processing.py` file.

**Review by skim0119 — APPROVED** [review]
LGTM

**Pull request merged**